applications/views.py

```diff
@@ -105,11 +106,11 @@ def applications_csv(request, city):
 
 
 @organiser_only
-def application_detail(request, city, app_id):
+def application_detail(request, city, app_number):
     """
     Display the details of a single application.
     """
-    application = Application.objects.get(pk=app_id)
+    application = [app for app in Application.objects.filter(number=app_number) if app.city == city][0]
```

Comment:
Ah, I didnt realize you can do `form__page__event__city=city` that nested!

I'll make the fix